pystats_docs.md

```diff
@@ -0,0 +1,81 @@
+# pystats documentation
+
+This describes the meaning of the various sections and fields in the pystats output.
+
+## Execution counts
+
+This is the counts of how many times each Tier 1 instruction is executed.
+
+The "miss ratio" column shows the percentage of times the instruction executed that it deoptimized.
+
+## Pair counts
+
+Shows the Top 100 most frequently occurring pairs of instructions as executed.
+
+## Predecessor / Successor pairs
+
+Shows the top 5 instructions that are executed before and after each instruction.
+
+## Specialization stats
+
+TBD
```

Comment:
Can we ask @markshannon tp help fill these TBD entries?

Reply:
For each family, there are three tables:

### Kind

Lists the number of "deferred" (i.e. not specialized) instructions executed, the number of "hits" (specialized instructions that complete) and "misses" (specialized instructions that deopt).

### Unnamed

Success: Number of specialization attempts that were successful
Failure: Number of specialization attempts that failed for some reason

### Failure kind

Numbers for the various kinds of specialization failures.
The total should add up to the "Failure" entry in the above table.

Reply:
@markshannon: For the Kind table, I see "specialization.deferred", "specialization.deopt", "hit" and "miss".  What is the difference between "deopt" and "miss"?

Reply:
@mdboom Can you add this to the doc?